src/main/java/org/example/filters/ResolveUrlTokenFilter.java

```diff
@@ -0,0 +1,62 @@
+package org.example.filters;
+
+import com.google.common.base.Preconditions;
+import org.apache.http.Header;
+import org.apache.http.HttpHeaders;
+import org.apache.http.HttpResponse;
+import org.apache.http.client.methods.HttpHead;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.apache.lucene.analysis.TokenFilter;
+import org.apache.lucene.analysis.TokenStream;
+import org.apache.lucene.analysis.tokenattributes.CharTermAttribute;
+
+import java.io.IOException;
+import java.util.regex.Pattern;
+
+public class ResolveUrlTokenFilter extends TokenFilter {
+    private final CharTermAttribute termAttribute = addAttribute(CharTermAttribute.class);
+    private static final Pattern patternToMatchShortenedUrls = Pattern.compile("https://bit.ly/\\w+");
```

Comment:
There's also a common practice to name constants in upper case (i.e. PATTERN_TO_MATCH_...)

Reply:
done